### PR TITLE
Expands `next/script` documentation

### PR DESCRIPTION
**Pull request by housseindjirdeh**
This PR does the following to partially address #31062:

- Expands the [Script Component](https://nextjs.org/docs/basic-features/script) page in the core documentation
- Adds a `next/script` API reference page

### CODE REVIEW

File: docs/basic-features/script.md [hunk 1]
```diff
@@ -13,102 +24,141 @@ description: Next.js helps you optimize loading third-party scripts with the bui
 
 </details>
 
-The Next.js Script component enables developers to set the loading priority of third-party scripts to save developer time and improve loading performance.
+The Next.js Script component, [`next/image`](/docs/api-reference/next/script.md), is an extension of the HTML `<script>` element. It enables developers to set the loading priority of third-party scripts to save developer time and improve loading performance.
```
Comment (leerob) [inline docs/basic-features/script.md:27]: Could we clarify how it helps save developer time?

File: docs/basic-features/script.md [hunk 2]
```diff
@@ -13,102 +24,141 @@ description: Next.js helps you optimize loading third-party scripts with the bui
...
-- `afterInteractive` (**default**): For scripts that can fetch and execute **after** the page is interactive, such as tag managers and analytics. These scripts are injected on the client-side and will run after hydration.
-- `lazyOnload` For scripts that can wait to load during idle time, such as chat support and social media widgets.
+Websites often use third-party scripts to include different types of functionality into their site, such as analytics, ads, customer support widgets, and consent management. However, this can introduce problems that impact both user and developer experience:
 
-> **Note:**
->
-> - `<Script>` supports inline scripts with `afterInteractive` and `lazyOnload` strategy.
-> - Inline scripts wrapped with `<Script>` _require an `id` attribute to be defined_ to track and optimize the script.
+- Some third-party scripts are heavy on loading performance and can drag down the user experience, especially if they are render-blocking
```
Comment (leerob) [inline docs/basic-features/script.md:45]: Maybe we link out to explain what "render blocking" is, or try to add a bit more to this sentence to explain

File: docs/basic-features/script.md [hunk 3]
```diff
@@ -13,102 +24,141 @@ description: Next.js helps you optimize loading third-party scripts with the bui
...
+
+There are two limitations to be aware of when using the Script component for inline scripts:
+
+- Only the `afterInteractive` and `lazyOnLoad` strategies can be used. **The `beforeInteractive` loading strategy cannot be used with inline scripts.**
+- An `id` attribute must be defined in order for Next.js to track and optimize the script
+
 ### Executing Code After Loading (`onLoad`)
 
-```js
+Some third-party scripts require users to run JavaScript code after the script has finished loading in order to instantiate content or fire a method. If you are loading a script with either `beforeInteractive` or `afterInteractive` as a loading strategy, you can execute code after it has loaded using the `onLoad` property:
```
Comment (leerob) [inline docs/basic-features/script.md:159]: I wonder what's the most common way to describe "fire a method" - maybe "call a function"?

File: docs/basic-features/script.md (hunk 1 above)
Comment (stefanprobst) [reply docs/basic-features/script.md:27]:
```suggestion
The Next.js Script component, [`next/script`](/docs/api-reference/next/script.md), is an extension of the HTML `<script>` element. It enables developers to set the loading priority of third-party scripts to save developer time and improve loading performance.
```

File: docs/basic-features/script.md [hunk 4]
```diff
@@ -13,102 +24,141 @@ description: Next.js helps you optimize loading third-party scripts with the bui
...
-  return (
-    <>
-      <Script
-        src="https://connect.facebook.net/en_US/sdk.js"
-        strategy="lazyOnload"
-      />
-    </>
-  )
-}
+#### lazyOnLoad
```
Comment (stefanprobst) [inline docs/basic-features/script.md:115]:
```suggestion
#### lazyOnload
```

File: docs/basic-features/script.md [hunk 5]
```diff
@@ -13,102 +24,141 @@ description: Next.js helps you optimize loading third-party scripts with the bui
...
+
+Scripts that use the `lazyOnLoad` strategy are loaded late after all resources have been fetched and during idle time. This strategy should be used for background or low priority scripts that do not need to load before or immediately after a page becomes interactive.
```
Comment (stefanprobst) [inline docs/basic-features/script.md:117]:
```suggestion
Scripts that use the `lazyOnload` strategy are loaded late after all resources have been fetched and during idle time. This strategy should be used for background or low priority scripts that do not need to load before or immediately after a page becomes interactive. 
```

File: docs/basic-features/script.md [hunk 6]
```diff
@@ -13,102 +24,141 @@ description: Next.js helps you optimize loading third-party scripts with the bui
...
+  id="show-banner"
+  dangerouslySetInnerHTML={{
+    __html: `document.getElementById('banner').classList.remove('hidden')`
+  }}
+/>
+```
+
+There are two limitations to be aware of when using the Script component for inline scripts:
+
+- Only the `afterInteractive` and `lazyOnLoad` strategies can be used. **The `beforeInteractive` loading strategy cannot be used with inline scripts.**
```
Comment (stefanprobst) [inline docs/basic-features/script.md:154]:
```suggestion
- Only the `afterInteractive` and `lazyOnload` strategies can be used. **The `beforeInteractive` loading strategy cannot be used with inline scripts.**
```

Comment (stefanprobst) [reply docs/basic-features/script.md:154]: should this mention *why* the `beforeInteractive` strategy does not allow inline scripts?

File: docs/basic-features/script.md (hunk 2 above)
Comment (housseindjirdeh) [reply docs/basic-features/script.md]: Added a few words to explain:

`...especially if they are render-blocking **and delay any page content from loading**`

File: docs/basic-features/script.md (hunk 3 above)
Comment (housseindjirdeh) [reply docs/basic-features/script.md]: Switched it to "call a function"

File: docs/basic-features/script.md (hunk 4 above)
Comment (housseindjirdeh) [reply docs/basic-features/script.md]: Fixed!

File: docs/basic-features/script.md (hunk 5 above)
Comment (housseindjirdeh) [reply docs/basic-features/script.md]: Fixed

File: docs/basic-features/script.md (hunk 6 above)
Comment (housseindjirdeh) [reply docs/basic-features/script.md]: Fixed, and explained in a little detail why `beforeInteractive` cannot be used

**Review by leerob — APPROVED** [review]

**Comment by leerob** [comment]
Much more clear - nice work!

File: docs/basic-features/script.md [hunk 7]
```diff
@@ -13,102 +24,135 @@ description: Next.js helps you optimize loading third-party scripts with the bui
...
+```jsx
+<Script src="https://connect.facebook.net/en_US/sdk.js" strategy="lazyOnload" />
 ```
 
-Now, you use `next/script` in the body of your Next.js page. It has client-side functionality that decides when and how to load the remote script based on the `strategy`.
+There are three different loading strategies that can be used:
 
-> **Note:**
->
-> - `next/script` **must not** be placed in either a `next/head` component or in `pages/_document.js`.
```
Comment (mclaros) [inline docs/basic-features/script.md:56]: Hi, was the Script component always able to be placed inside of `next/head` of `_document.js`? Or was this achieved via a change to the Script component in a recent release?

**Pull request merged**